module3/lessons/Week4/UpdatingResources.md

```diff
@@ -0,0 +1,219 @@
+{
+    // Arrange
+    var context = GetDbContext();
+    var client = _factory.CreateClient();
+
+    Movie movie = new Movie { Title = "Spaceballs", Genre = "Comedy" };
+    context.Movies.Add(movie);
+    context.SaveChanges();
+
+    // Act
+    var response = await client.GetAsync($"/movies/{movie.Id}/edit");
+    var html = await response.Content.ReadAsStringAsync();
+
+    // Assert
+    Assert.Contains("Edit Movie", html);
+    Assert.Contains(movie.Title, html);
+    Assert.Contains(movie.Genre, html);
+}
+```
+
+</section>
+
+**Code-Along**  
+Now let's build the action that will make this test pass!
+
+### HTTP and REsTful Routes
+
+You will note in our walkthrough that even though the most **ReST**ful way to send an update request is with a `PUT` method; but, we are using `POST` in our form - why?
+
+Because technology is changing all the time, we often have to work around differences in legacy or foundational structures (like HTML), and newer design strategies (like ReST).  
+
+HTML has been used since the early 1990s.  At that time, the internet was a much simpler place.  The form tags we still use are designed to work with only two Http methods: Get and Post.
+
+ReST was defined in the early 2000's as a way to standardized http requests.  As the internet became more complex, we started to see requests being made by applications and not just by users.  Applications could more easily create ReSTful routes and requests without using a user interface (we will learn more about this in Mod4!).  
+
+<section class='instructor-notes' markdown='1'>
+Walk through the implementation together - this would be a good opportunity to let students help drive the implementation!
+</section>
+
+
+<section class='call-to-action' markdown='1'>
+
+**With a Partner**  
+Add a test to our MoviesControllerTests that satisfies this user-story
+```
+As a User
+When I visit '/movies/1/edit'
+When I update the Title and/or Genre
+    And click 'Update Movie'
+Then I am redirected to the Movie Show Page
+    And I see the updated movie information
+```
+
+</section>
+
+
+<section class='answer' markdown='1'>
+
+### One Solution
+```c#
+[Fact]
+public async Task Update_SavesChangesToMovie()
+{
+    // Arrange
+    var context = GetDbContext();
+    var client = _factory.CreateClient();
+
+    Movie movie = new Movie { Title = "Goofy", Genre = "Comedy" };
+    context.Movies.Add(movie);
+    context.SaveChanges();
+
+    var formData = new Dictionary<string, string>
+    {
+        { "Title", "Goofy" },
+        { "Genre", "Documentary" }
+    };
+
+    // Act
+    var response = await client.PostAsync($"/movies/{movie.Id}", new FormUrlEncodedContent(formData));
+    var html = await response.Content.ReadAsStringAsync();
+
+    // Assert
+    response.EnsureSuccessStatusCode();
+    Assert.Contains("Goofy", html);
+    Assert.Contains("Documentary", html);
+    Assert.DoesNotContain("Comedy", html);
+}
+```
+
+</section>
+
+### the Order of Our Tests
```

Comment:
I copy pasted this entire section into my Create Single Resource Lesson.

Reply:
Great! I'll remove it from this lesson :)